File: back_end/src/main/java/com/wireless/model/entity/AiotDevice.java
package com.wireless.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

/**
 * AIOT 无线传感终端设备实体
 * 对应 ESP32 上报的全部字段
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AiotDevice {
    private Long id;
    private String deviceCode;      // 设备唯一编码
    private String deviceName;      // 设备名称
    private String deviceType;      // esp32/sensor/gateway
    private String ipAddress;       // IP 地址
    private String macAddress;      // MAC 地址
    private String mqttTopic;       // MQTT 上报主题
    private String firmwareVersion; // 固件版本
    private String sensorList;      // 传感器清单 JSON

    // ESP32 上报的实时状态
    private Integer ledStatus;      // LED 0-关 1-开
    private String mode;            // auto/manual
    private Integer lightSensor;    // 光敏值
    private Integer pirStatus;      // 人体感应 0-无人 1-有人
    private Integer screenStatus;   // 屏幕 0-息屏 1-亮屏
    private Integer wifiRssi;       // WiFi 信号
    private Integer tcpConnected;   // TCP 连接

    // 管理字段
    private Integer onlineStatus;   // 0-离线 1-在线
    private LocalDateTime lastReportTime;
    private String registerSource;  // manual/auto
    private String remarks;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: Android软件/app/src/main/java/com/smarthome/app/core/model/AiotDevice.kt
package com.smarthome.app.core.model

import com.squareup.moshi.Json

data class AiotDevice(
    @Json(name = "id") val id: Long = 0,
    @Json(name = "deviceCode") val deviceCode: String = "",
    @Json(name = "deviceName") val deviceName: String = "",
    @Json(name = "deviceType") val deviceType: String? = null,
    @Json(name = "ipAddress") val ipAddress: String? = null,
    @Json(name = "macAddress") val macAddress: String? = null,
    @Json(name = "mqttTopic") val mqttTopic: String? = null,
    @Json(name = "firmwareVersion") val firmwareVersion: String? = null,
    @Json(name = "sensorList") val sensorList: String? = null,
    @Json(name = "ledStatus") val ledStatus: Int? = null,
    @Json(name = "mode") val mode: String? = null,
    @Json(name = "lightSensor") val lightSensor: Int? = null,
    @Json(name = "pirStatus") val pirStatus: Int? = null,
    @Json(name = "screenStatus") val screenStatus: Int? = null,
    @Json(name = "wifiRssi") val wifiRssi: Int? = null,
    @Json(name = "tcpConnected") val tcpConnected: Boolean? = null,
    @Json(name = "onlineStatus") val onlineStatus: Boolean? = null,
    @Json(name = "lastReportTime") val lastReportTime: String? = null,
    @Json(name = "registerSource") val registerSource: String? = null,
    @Json(name = "remarks") val remarks: String? = null,
    @Json(name = "createTime") val createTime: String? = null,
    @Json(name = "updateTime") val updateTime: String? = null
)
